src/hooks/usePncWebSocket.ts

```diff
@@ -0,0 +1,81 @@
+import { useEffect } from 'react';
+
+import * as pncWebSocketService from 'services/pncWebSocketService';
+
+/**
+ * HOOK CHECKERS
+ */
+
+/**
+ * Additional filtering parameters
+ */
+interface IOnParameters {
+  buildConfigId?: string;
+  userId?: string;
+}
+
+/**
+ * Check whether Build status change WebSocket event was sent.
+ *
+ * @param wsData - WebSocket data
+ * @param parameters - See {@link IOnParameters}
+ * @returns true when Build status changed, otherwise false
+ */
+export const onBuildStatusChanged = (wsData: any, { buildConfigId, userId }: IOnParameters = {}) => {
+  let result = wsData.job === 'BUILD' && wsData.notificationType === 'BUILD_STATUS_CHANGED';
+  if (result && buildConfigId) {
+    result = result && wsData.build.buildConfigRevision.id === buildConfigId;
+  }
+  if (result && userId) {
+    result = result && wsData.build.user.id === userId;
+  }
+  return result;
+};
+
+/**
+ * Check whether new Build started WebSocket event was sent.
+ *
+ * @param wsData - WebSocket data
+ * @param parameters - See {@link IOnParameters}
+ * @returns true when new Build was started, otherwise false
+ */
+export const onBuildStarted = (wsData: any, { buildConfigId, userId }: IOnParameters = {}) =>
+  onBuildStatusChanged(wsData, { buildConfigId, userId }) && wsData.oldStatus === 'NEW';
+
+/**
+ * HOOK
+ */
+
+interface IUsePncWebSocketEffectOptions {
+  // condition based listening, when true, events will be ignored (for example when Build is finished)
+  preventListening?: boolean;
```

Comment:
This may be a misleading description when compared to implementation. The implementation stops callback from being registered. But when the condition changes, it will remain registered.

Reply:
When the condition changes, `return` will be executed